std/math/operations.d

```diff
@@ -1717,7 +1717,7 @@ if (isFloatingPoint!T)
     bool negative;
 }
 
-FloatingPointBitpattern!T extractBitpattern(T)(T val)
+FloatingPointBitpattern!T extractBitpattern(T)(T val) @trusted
```

Comment:
please leave the trusted lambdas in place and remove this

Reply:
It looks like `ldexp` is not @safe. Anyway, having such functions marked as `@trusted` is bad practice.

Reply:
`core.math.ldexp` is @safe. I get eye cancer from these trusted lambdas and prefer an @trusted for the whole function instead of 4 lambdas, especially for such a trivial function (except for the CTFE part) with the grand task of splitting the packed bits into 3 integral components, with the obvious need to reinterpret the FP value as integer.

Reply:
Maybe I am missing something, but it [seems](https://github.com/dlang/druntime/blob/master/src/core/math.d#L106) that `ldexp` is not **@safe** annotated.

I agree that the lambdas don't look very well, but they have the added benefit of reducing the search area when something goes wrong. Otherwise, someone unfamiliar with the code needs to scan entire function (~100 LOC). Even if you find the code trivial, that might not be the case for other developers. Also, the phobos convention is to reduce the trusted area to a minimum.

Reply:
>Maybe I am missing something

https://github.com/dlang/druntime/blob/46bca630e501f01ce78124220d766e0f643547b0/src/core/math.d#L31`

> Otherwise, someone unfamiliar with the code needs to scan the entire function (~100 LOC).

Scan it for what?

---

The lambdas have been removed in their separate commit, so feel free to rework if you guys insist on them. I'm tired of debating such stuff.

Reply:
> Scan for what?

In case some safety issue appears.

Reply:
I don't really follow - why would safety issues appear? They could have appeared previously with the function being @safe, and some called function (in the CTFE part, the runtime part doesn't call any function) changing to unsafe. Extracting integral sign, exponent and mantissa from an FP value should obviously be safe; @trusted just slightly simplifies the implementation wrt. the used unsafe reinterpret-casts.

Similar functions operating on the FP bits in [std.math.traits](https://github.com/dlang/phobos/blob/master/std/math/traits.d) (signbit, isIdentical, isInfinity, isSubnormal, ...) are all @trusted.